src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -1181,6 +1182,82 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
         p.get_processing_order().calc_processing_order(p);
 }
 
+void prepare_primitive_fusing::fuse_constant_transposes(program& p) {
+    std::function<bool(const cldnn::program_node*)> is_weights_path =
+        [&is_weights_path](const cldnn::program_node* node) {
+        if (node->get_users().empty())
+            return false;
+
+        const auto* next_node = node->get_users().front();
+
+        if (next_node->is_type<fully_connected>() ||
+            next_node->is_type<deconvolution>() ||
+            next_node->is_type<convolution>() ||
+            next_node->is_type<binary_convolution>() ||
+            next_node->is_type<deformable_conv>()) {
+            size_t weights_offset = next_node->get_primitive()->input_size();
+            return &next_node->get_dependency(weights_offset) == node;
+        }
+
+        if (node->is_constant() && node->get_users().size() == 1)
+            return is_weights_path(next_node);
+
+        return false;
+    };
+
+    auto convert_data_format_by_order = [](format fmt, const std::vector<uint16_t>& order) -> format {
+        const auto& old_order = fmt.dims_order();
+        auto new_order = old_order;
+
+        for (size_t i = 0; i < order.size(); ++i) {
+            new_order[i] = old_order[order[i]];
+        }
+
+        return format::find_format(new_order, fmt.block_sizes());
+    };
+
+    auto itr = p.get_processing_order().begin();
+    while (itr != p.get_processing_order().end()) {
+        auto& node = *itr++;
+
+        if (!node->is_type<permute>())
+            continue;
+
+        auto& permute_node = node->as<permute>();
+
+        if (!is_weights_path(&permute_node) || !permute_node.get_dependency(0).is_type<data>())
+            continue;
+
+        auto& prev_const = permute_node.get_dependency(0).as<data>();
+
+        if (prev_const.get_users().size() != 1)
+            continue;
+
+        const auto& permute_order = permute_node.get_primitive()->permute_order;
+
+        format new_fmt = format::any;
+        try {
+            new_fmt = convert_data_format_by_order(prev_const.get_output_layout().format, permute_order);
+        } catch(ov::Exception&) {
+            continue;
+        }
+
+        layout updated_const_layout = prev_const.get_output_layout();
+        updated_const_layout.format = new_fmt;
+        updated_const_layout.set_partial_shape(permute_node.get_output_pshape());
+
+        p.extract_and_remove(permute_node);
+
+        const auto& new_mem = p.get_engine().reinterpret_buffer(prev_const.get_attached_memory(), updated_const_layout);
+
+        auto new_const_prim = std::make_shared<data>(prev_const.id() + "_fused_with_transpose", new_mem);
+        auto& new_const_node = p.get_or_create(new_const_prim);
+
+        p.replace(prev_const, new_const_node);
+        new_const_node.recalc_output_layout(false);
+    }
+}
```

Comment:
I'd like to be aligned with the basic idea,
Is my understanding correct? 
- The basic idea is to faking constant which is not actually transposed, but setting the expected transposed layout. 
- And the actual transpose will be done at the post weight optimization 

However I have curiosity can this be applied to random transpose order? 
Can this be applied for orders like (2, 1, 3, 0) ?
Can't it be applied to a transpose with only two dimensions exchanging? (e.g., 0, 1, 3, 2)

Reply:
It won't work for some combinations. But if the format is not found for given permutation, then such transpose will not be merged and will be folder later.